### PR TITLE
Add several speedup examples

### PR DESCRIPTION
**Pull request by zheng-ningxin**
Add a few common detection model examples (if the model written by the user fails to speedup, he/she can refer to the repo we gave).

### CODE REVIEW

File: examples/model_compress/pruning/speedup/speedup_yolov3.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import torch
+from pytorchyolo import models
+
+from nni.compression.pytorch import ModelSpeedup
+from nni.algorithms.compression.pytorch.pruning import L1FilterPruner, LevelPruner
+from nni.compression.pytorch.utils import not_safe_to_prune
```
Comment (J-shang) [inline examples/model_compress/pruning/speedup/speedup_yolov3.py:6]: seems this function has not updated.

Comment (zheng-ningxin) [reply examples/model_compress/pruning/speedup/speedup_yolov3.py:6]: Yes, this pr depends on https://github.com/microsoft/nni/pull/3462

Comment (J-shang) [reply examples/model_compress/pruning/speedup/speedup_yolov3.py:6]: got it

**Comment by linbinskn** [comment]
The file **model_speedup.py** can't be moved directly to the new folder speedup since
> https://github.com/microsoft/nni/blob/master/examples/model_compress/pruning/model_speedup.py#L10

File: examples/model_compress/pruning/speedup/speedup_nanodet.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+from nanodet.util import cfg, load_config                                                                                                                                                                                                                                                                  
+
+from nni.compression.pytorch import ModelSpeedup
+from nni.algorithms.compression.pytorch.pruning import L1FilterPruner
+
+"""
+NanoDet model can be installed from https://github.com/RangiLyu/nanodet.git
+"""
+
+cfg_path = r"nanodet/config/nanodet-RepVGG-A0_416.yml"
```
[the remark this answers is not in the document]
Comment (QuanluZhang) [reply examples/model_compress/pruning/speedup/speedup_nanodet.py:12]: no need to put their code into our repo. but we can provide concrete commands about how to prepare the code in the comment

File: examples/model_compress/pruning/speedup/speedup_yolov3.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  "%s/yolov3.weights" % prefix)
+model.eval()
+dummy_input = torch.rand(8, 3, 320, 320)
+model(dummy_input)
+# Generate the config list for pruner
+# Filter the layers that may not be able to prune
+not_safe = not_safe_to_prune(model, dummy_input)
+cfg_list = []
+for name, module in model.named_modules():
+  if name in not_safe:
```
Comment (QuanluZhang) [inline examples/model_compress/pruning/speedup/speedup_yolov3.py:22]: maybe we can leverage "exclude" here in the config, @J-shang

Comment (zheng-ningxin) [reply examples/model_compress/pruning/speedup/speedup_yolov3.py:22]: Yes, `exclude` is much more elegant.

**Review by QuanluZhang — APPROVED** [review]

**Review by J-shang — APPROVED** [review]

**Review by linbinskn — APPROVED** [review]

**Pull request merged**